### ISSUE TITLE
vararg tuple types not working when vararg not at last place

### ISSUE THREAD

**Issue by mauro3** [JuliaLang/julia#10770]
This does not work:
```
julia> (Int,Int,Float64)<:(Int...,Float64)
ERROR: type: subtype: expected Type{T<:Top}, got (DataType,DataType)
```
however inside `methods` a tuple like that does seem to do something, at least when a `Any...` is involved:
```
julia> methods(f, (Any...,Int,IOBuffer))
1-element Array{Any,1}:
 f(a::Int64,b::Int64,c::Float64) at none:1

julia> methods(f, (Int...,Float64))
0-element Array{Any,1}
```
I'm not sure how it should work but probably not as above. Allowing several `...` inside a type tuple could be useful.  Related:  #10380, #8974

**Comment by JeffBezanson** [comment]
This is absolutely not supported yet. It would be very useful, but extremely difficult. The weirdness with `methods` is kind of a syntax bug that will be fixed by #10380; after that it won't be possible to write that.

**Comment by timholy** [comment]
By and large there is zero support for varargs anywhere except the last slot. Many of us have had times where it could be useful, but having dug into the code that supports varargs I agree that it would represent an enormous increase in complexity.

I suspect that a good number of things we do with varargs now will be done with tuples in the future. See, for example, the discussion in #10691. So I don't think it's worth going to the effort to implement this, certainly not until we see how the tuple revamp works out.

**Comment by mauro3** [comment]
In that case, maybe throw an error on construction?

**Comment by JeffBezanson** [comment]
Yes, that's what will happen, because `(Int..., )` will mean to splice `Int` into a tuple.

**Comment by JeffBezanson** [comment]
fixed by #10380
